python/lsst/drp/tasks/reprocess_visit_image.py

```diff
@@ -0,0 +1,498 @@
+        psf,
+        background,
+        ap_corr,
+        photo_calib,
+        wcs,
+        summary_stats,
+        id_generator=None,
+        result=None
+    ):
+        """Detect and measure sources on the exposure(s) (snap combined as
+        necessary), and make a "final" Processed Visit Image using all of the
+        supplied metadata, plus a catalog measured on it.
+
+        Parameters
+        ----------
+        exposures : `lsst.afw.image.Exposure` or
+                    `list` [`lsst.afw.image.Exposure`]
+            Post-ISR exposure(s), with an initial WCS, VisitInfo, and Filter.
+            Modified in-place during processing if only one is passed.
+            If two exposures are passed, treat them as snaps and combine
+            before doing further processing.
+        psf : `lsst.afw.detection.Psf`
+            PSF model for this exposure.
+        background : `lsst.afw.math.BackgroundList`
+            Total background that had been fit to the exposure so far;
+            modified in place to include background fit when detecting sources.
+        ap_corr : `lsst.afw.image.ApCorrMap`
+            Aperture Correction model for this exposure.
+        photo_calib : `lsst.afw.image.PhotoCalib`
+            Photometric calibration model for this exposure.
+        wcs : `lsst.afw.geom.SkyWcs`
+            World Coordinate System model for this exposure.
+        summary_stats : `lsst.afw.image.ExposureSummaryStats`
+            Summary statistics measured on this exposure.
+        id_generator : `lsst.meas.base.IdGenerator`, optional
+            Object that generates source IDs and provides random seeds.
+        result : `lsst.pipe.base.Struct`, optional
+            Result struct that is modified to allow saving of partial outputs
+            for some failure conditions. If the task completes successfully,
+            this is also returned.
+
+        Returns
+        -------
+        result : `lsst.pipe.base.Struct`
+            Results as a struct with attributes:
+
+            ``exposure``
+                Calibrated exposure, with pixels in nJy units.
+                (`lsst.afw.image.Exposure`)
+            ``source``
+                Sources that were measured on the exposure calibrated fluxes
+                and magnitudes.
+                (`astropy.table.Table`)
+            ``source_footprints``
+                Footprints of sources that were measured on the exposure.
+                (`lsst.afw.table.SourceCatalog`)
+        """
+        if result is None:
+            result = pipeBase.Struct()
+        if id_generator is None:
+            id_generator = lsst.meas.base.IdGenerator()
+
+        result.exposure = self.snap_combine.run(exposures).exposure
+
+        # Calibrate the image, so it's on the same units as the background.
+        result.exposure.maskedImage = photo_calib.calibrateImage(result.exposure.maskedImage)
+        result.exposure.maskedImage -= background.getImage()
```

Comment:
This may have been discussed elsewhere, but it's all fuzzy.  You say "so it's the same units as the background" but where is the background put into calibrated units?

Reply:
You and I should maybe talk about that, but on [DM-45780](https://rubinobs.atlassian.net/browse/DM-45780) I made the background output by `calibrateImage` be in nJy, so that you can add it to the `initial_pvi` to get the un-subtracted image. Otherwise, it's in ADU but `initial_pvi` is in nJy, so they're not directly comparable.